FirebasePhoneAuthUI/CountryCode/FUICountryCodes.m

```diff
@@ -163,19 +154,58 @@ - (FUICountryCodeInfo *)countryCodeInfoForPhoneNumber:(NSString *)phoneNumber {
   return nil;
 }
 
-- (FUICountryCodes *)searchCountriesByName:(NSString *)countryName {
-  if ([countryName length] == 0) {
-    return nil;
-  } else {
-    NSArray *results =
-        [self.countryCodesArray filteredArrayUsingPredicate:
-            [NSPredicate predicateWithFormat:FUI_JSON_COUNTRY_NAME_PREDICATE, countryName]];
-    return [[FUICountryCodes alloc] initWithCountriesArray:results];
+- (void)blackListCountries:(NSSet<NSString *> *)countries {
+  NSMutableArray<NSDictionary *> *array =
+      [[NSMutableArray alloc] initWithCapacity:_countryCodesArray.count];
+  for (NSDictionary *dict in self.countryCodesArray) {
+    NSString *countryCode = dict[FUI_JSON_COUNTRY_CODE_KEY];
+    NSString *dialCode = dict[FUI_JSON_DIALCODE_KEY];
+    if ([countries containsObject:countryCode] || [countries containsObject:dialCode]) {
+      continue;
```

Comment:
Is it worth raising an error here? If so, should it be a fatal or non-fatal error?

Reply:
The design doc specifically said we should ignore this if developers gives invalid code, so it's on par with android and web implementation. 

I'll add a debug only assert.